esp/clients/ws_dfsclient/ws_dfsclient.cpp

```diff
@@ -556,16 +561,64 @@ IClientWsDfs *getDfsClient(const char *serviceUrl, IUserDescriptor *userDesc)
     return dfsClient.getClear();
 }
 
+static CriticalSection localSecretCrit;
+static void configureClientSSL(IEspClientRpcSettings &rpc, const char *secretName)
```

Comment:
Another possibility would be to add a new method to jsecrets:
```
void setRpcClientSecretInfo(IEspClientRpcSettings &rpc, const char *category, const char *secret);
```
That for now, sets up the client rpc settings with the actual paths to the secrets.  Later once I add binary blob support to IEspClientRpcSettings I can change the routine to set them using a blob.

Reply:
similarly, my 1st cut off this workaround, added secret methods to get the direct path to the K8s secrets, and then use them directly (to avoid the save to local), but decided against it in the end, because it would mean you would not be able to use Vault secrets.
Ok, so a jsecrets method could detect where the secrets were coming from (K8s or Vault), and only have to save them to local if Vault, but as this is required for vault secrets anyway, and it's temporary/will be removed once HPCC-27255 is done, I think ok as is.

Reply:
Agreed.  Yes, I'm definitely not a huge fan of calling jsecrets to get the paths... it seems to miss the point somehow and doing so through an interface as I described isn't that much better.